sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1755,4 +1755,27 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
       cached.unpersist()
     }
   }
+
+  test("SPARK-46037: When Left Join build Left, ShuffledHashJoinExec may " +
+    "result in incorrect results") {
+    withSQLConf(SQLConf.ENABLE_BUILD_SIDE_OUTER_SHUFFLED_HASH_JOIN_CODEGEN.key -> "false") {
+      val df1 = sql(
+        """
+          |SELECT /*+ SHUFFLE_HASH(t1) */ *
```

Comment:
In the golden file test `outer-join.sql`, we run it multiple times to test different code paths: broadcast join, shuffle join, sort merge join. code on/off etc., can we move the test there?